docs/src/reference/asciidoc/core/kerberos.adoc

```diff
@@ -0,0 +1,481 @@
+specific configurations are set. It's usually sufficient to do it right before submitting the job. You should only do
+this once per job since each call will wastefully obtain another API Key.
+
+Additionally, the utility is also compatible with the `mapred` API classes:
+
+[source,java]
+----
+JobConf jobConf = new JobConf(getConf()); <1>
+jobConf.setJobName("My-Job-Name");
+
+// Configure JobConf Here...
+
+EsMapReduceUtil.initCredentials(jobConf); <2>
+
+JobClient.runJob(jobConf).waitForCompletion(); <3>
+----
+<1> Creating a new job configuration
+<2> Obtain {es} delegation tokens
+<3> Submit the job to the cluster
+
+[[kerberos-hive]]
+=== {krb} on Hive
+
+[[kerberos-hive-requirements]]
+==== Requirements
+
+Before using {krb} authentication to {es} in Hive, Hadoop Security must be enabled. Make sure you have done all the
+required steps for <<kerberos-hadoop-requirements,configuring your Hadoop cluster>> for use with {es} and {krb}.
+
+Using {krb} auth on {es} via {eh} and Hive is only supported using HiveServer2.
+
+Finally, ensure that Hive Security is enabled.
+
+Since Hive relies on user impersonation in {es} it is advised that you familiarise yourself with
+https://www.elastic.co/guide/en/elastic-stack-overview/current/setting-up-authentication.html[{es} authentication] and
+https://www.elastic.co/guide/en/elastic-stack-overview/current/authorization.html[authorization].
+
+[[kerberos-hive-requirements-proxy]]
+===== Configure user impersonation settings for Hive
+
+Hive's security model follows a proxy-based approach. When a client submits a query to a secured Hive server, Hive
+authenticates the client using {krb}. Once Hive is sure of the client's identity, it wraps its own identity with a
+_proxy user_. The proxy user contains the client's simple user name, but contains no credentials. Instead, it is
+expected that all interactions are executed as the Hive principal impersonating the client user. This is why when
+configuring Hive security, one must specify in the Hadoop configuration which users Hive is allowed to impersonate:
+
+[source,xml]
+----
+<property>
+    <name>hadoop.proxyuser.hive.hosts</name>
+    <value>*</value>
+</property>
+<property>
+    <name>hadoop.proxyuser.hive.groups</name>
+    <value>*</value>
+</property>
+----
+
+{es} https://www.elastic.co/guide/en/elastic-stack-overview/current/run-as-privilege.html[supports user impersonation],
+but only users from certain realm implementations can be impersonated. Most deployments of {krb} include other identity
+management components like
+https://www.elastic.co/guide/en/elasticsearch/reference/current/configuring-ldap-realm.html[LDAP] or
+https://www.elastic.co/guide/en/elasticsearch/reference/current/configuring-ad-realm.html[Active Directory].
+In those cases, you can configure those realms in {es} to allow for user impersonation.
+
+If you are only using {krb}, or you are using a solution for which {es} does not support user impersonation, you must
```

Comment:
Could you explain why this process is so complicated?

Reply:
As mentioned above, The Hive principal is the one that performs all the authentication to external services, and needs to be allowed to impersonate the user that submitted the query. Kerberos realms do not support impersonating their users in Elasticsearch, so the principals from the Kerberos realm need to be mirrored to an Elasticsearch realm that _does_ support user impersonation (like file, native, LDAP, or Active Directory).

I can see about making this simpler in the docs above.